test/topology_random_failures/cluster_events.py

```diff
@@ -0,0 +1,624 @@
+    await manager.api.move_tablet(
+        node_ip=servers[0].ip_addr,
+        ks="test",
+        table="test",
+        src_host=old_host,
+        src_shard=old_shard,
+        dst_host=new_host,
+        dst_shard=0,
+        token=0,
+    )
+
+    yield
+
+    replicas = await get_all_tablet_replicas(
+        manager=manager,
+        server=servers[0],
+        keyspace_name="test",
+        table_name="test",
+    )
+
+    assert len(replicas) == 1
+
+    hosts_with_replica = [r[0] for r in replicas[0].replicas]
+
+    assert len(hosts_with_replica) == 3
+    assert new_host in hosts_with_replica
+    assert old_host not in hosts_with_replica
+
+
+@deselect_for(
+    error_injections=[
+        "stop_after_starting_auth_service",
+        "stop_after_setting_mode_to_normal",
+        "stop_before_becoming_raft_voter",
+        "stop_after_updating_cdc_generation",
+        "stop_before_streaming",
+        "stop_after_streaming",
+    ],
+    reason="Can't add a node to a cluster with a banned node",
+)
+async def remove_data_dir_of_dead_node(manager: ManagerClient,
+                                       random_tables: RandomTables,
+                                       error_injection: str) -> AsyncIterator[None]:
+    running_servers = await manager.running_servers()
+    data_dir = os.path.join(await manager.server_get_workdir(running_servers[1].server_id), "data")
+
+    LOGGER.info("Kill a node")
+    await manager.server_stop(server_id=running_servers[1].server_id)
+    await manager.server_not_sees_other_server(
+        server_ip=running_servers[0].ip_addr,
+        other_ip=running_servers[1].ip_addr,
+    )
+
+    yield
+
+    LOGGER.info("Remove data dir of the dead node and start it")
+    rmtree(path=data_dir, ignore_errors=True)
+    await manager.server_start(server_id=running_servers[1].server_id)
+
+    yield
+
+
+@deselect_for(
+    error_injections=[
+        "stop_after_starting_auth_service",
+        "stop_after_setting_mode_to_normal",
+        "stop_before_becoming_raft_voter",
+        "stop_after_updating_cdc_generation",
+        "stop_before_streaming",
+        "stop_after_streaming",
+    ],
+    reason="See issue #18640 (failed to add a node to a cluster if another bootstrapping node is stuck)",
+)
+async def add_new_node(manager: ManagerClient,
+                       random_tables: RandomTables,
+                       error_injection: str) -> AsyncIterator[None]:
+    yield
+
+    LOGGER.info("Add a new node to the cluster")
+    await manager.server_add(timeout=TOPOLOGY_TIMEOUT)
+
+    yield
+
+
+@deselect_for(
+    error_injections=[
+        "stop_after_starting_auth_service",
+        "stop_after_setting_mode_to_normal",
+        "stop_before_becoming_raft_voter",
+        "stop_after_updating_cdc_generation",
+        "stop_before_streaming",
+        "stop_after_streaming",
+    ],
+    reason="See issue #19151 (decommission process stuck while bootstrapping node is paused)",
```

Comment:
As above

Reply:
We decided to leave it outside of this PR.